### PR TITLE
Bump rn to the newest v

### PR DESCRIPTION
**Pull request by osdnk**
Fixes RNBW-2281

## What changed (plus any additional context for devs)

Updating to the new RN.

The problem was that now we use `Gradle-7.3.3`, what's significantly different due to the deprecation of the `maven` packager and the `compile` command. Therefore, I bumped some packages and modified `build.gradle` of a few others (that were not updated by maintainers) 

## PoW (screenshots / screen recordings)

n/a

## Dev checklist for QA: what to test

check if apps compile. You might need to reset the cache, remove `node_modules` etc.

Everything might be broken. Especially, things like animations, alerts, random text misalignments, navigation or even random libraries' behavior 

NEEDS FULL REGRESSION

TestFlight build: 1.6.10 (19) 
Bitrise will fail until: https://github.com/rainbow-me/rainbow-scripts/pull/74/files

### CODE REVIEW

**Review by tchayen — APPROVED** [review]
Checked on iOS and Android. So far everything works the same.

File: src/hooks/usePersistentAspectRatio.ts [hunk 1]
```diff
@@ -16,7 +16,7 @@ enum State {
 
 type Result = {
   state: State;
-  result: number;
+  result: number | undefined;
```
Comment (tchayen) [inline src/hooks/usePersistentAspectRatio.ts:19]: Is this in scope of this PR?

Comment (osdnk) [reply src/hooks/usePersistentAspectRatio.ts:19]: Yes! Updating RN => Updating MMKV => issues with typings!

File: android/app/build.gradle [hunk 2]
```diff
@@ -207,7 +210,36 @@ android {
...
+                        "NDK_TOOLCHAIN_VERSION=clang",
+                        "GENERATED_SRC_DIR=$buildDir/generated/source",
+                        "PROJECT_BUILD_DIR=$buildDir",
+                        "REACT_ANDROID_DIR=$rootDir/../node_modules/react-native/ReactAndroid",
+                        "REACT_ANDROID_BUILD_DIR=$rootDir/../node_modules/react-native/ReactAndroid/build"
+                    cFlags "-Wall", "-Werror", "-fexceptions", "-frtti", "-DWITH_INSPECTOR=1"
+                    cppFlags "-std=c++17"
+                    // Make sure this target name is the same you specify inside the
+                    // src/main/jni/Android.mk file for the `LOCAL_MODULE` variable.
+                    targets "rndiffapp_appmodules"
```
Comment (brunobar79) [inline android/app/build.gradle:229]: are you sure about this line? isn't `rndiffapp` the default app name from [React Native Upgrade Helper](https://react-native-community.github.io/upgrade-helper) ?

File: android/app/src/main/java/me/rainbow/MainApplication.java [hunk 3]
```diff
@@ -26,11 +26,11 @@
 import me.rainbow.NativeModules.RNTextAnimatorPackage.RNTextAnimatorPackage;
 import me.rainbow.NativeModules.RNZoomableButton.RNZoomableButtonPackage;
 import com.microsoft.codepush.react.CodePush;
+import com.facebook.react.config.ReactFeatureFlags;
 
 class RainbowJSIModulePackage extends ReanimatedJSIModulePackage {
     @Override
     public List<JSIModuleSpec> getJSIModules(ReactApplicationContext reactApplicationContext, JavaScriptContextHolder jsContext) {
-        MmkvModule.install(jsContext, reactApplicationContext.getFilesDir().getAbsolutePath() + "/mmkv");
```
Comment (brunobar79) [inline android/app/src/main/java/me/rainbow/MainApplication.java:33]: If you're removing this, you should remove [this import ](https://github.com/rainbow-me/rainbow/pull/3245/files#diff-f5b10a38d3df5e856664549745a6e4aa77e7e35a81a7e84b01559472870729c9R17)too

**Review by brunobar79 — APPROVED** [review]
LGTM!  

@ibrahimtaveras00  make sure you peep the Walletconnect flow (automatic redirects back to the app)

**Review by jkadamczyk — APPROVED** [review]

**Review by terrysahaidak — APPROVED** [review]

**Comment by osdnk** [comment]
\rebase

**Review by jinchung — APPROVED** [review]
🌮 

pending full regression approval

**Comment by ibrahimtaveras00** [comment]
Did a full run through on both OS's.  I have the latest fixes on Android and can verify that I can hide/pin tokens on the wallet and also no white screen.  I have to do another testflight build on this (after version bump on develop) to verify those two fixes on iOS as I can't build locally on sim or with xcode on this branch.

**Review by ibrahimtaveras00 — APPROVED** [review]
Build on iOS is working now and I was able to test the fixes, QA Passed 👍🏽
